test/test_examples.py

```diff
@@ -442,6 +442,62 @@ def test_cuse(output_checker):
     finally:
         mount_process.terminate()
 
+def test_746(tmpdir, output_checker):
+    """test case for Issue #746
+
+    If RELEASE and UNLINK opcodes are sent back to back, and fuse_fs_release()
+    and fuse_fs_rename() are slow to execute, UNLINK will run while RELEASE is
+    still executing. UNLINK will try to rename the file and, while the rename
+    is happening, the RELEASE will finish executing. As a result, RELEASE will
+    not detect in time that UNLINK has happened, and UNLINK will not detect in
```

Comment:
To test this, do we need a passthrough filesystem at all? Can't we have something super simple (along the lines of hello.c) that tests *in the filesystem* that things are happening in the right order?

Reply:
I guess it might be able to work. I assume it will need a bit of an elaborate test to emulate the creation, renaming, and deletion of multiple files, listdirs, etc, and be `nullpath_ok` compliant. This itself will need validation as well, and future API maintenance.

This test started with the LD_PRELOAD as a proper end-to-end black-box testing of the intermediate layers through the use of the examples that should always represent the correct and expected design of a FUSE application, and with the potential to expand to other examples and to other type of fault/latency-injection and dynamic testing.
Creating a fake FS implementation with a very limited functionality and pre-coded behaviour invalidates the whole point of black-box testing.
In this case the whole test could be scraped and might be able to be replaced with an atomic flag and an assert within `fuse.c` to check that `release()` and `unlink()` do not run simultaneously.

Anyone offering to write and validate the custom FS implementation for this test?